Sources/HAP/Server/Device.swift

```diff
@@ -47,9 +47,81 @@ public class Device {
     public let setupCode: String
     let storage: Storage
     let pairings: Pairings
-    public let accessories: [Accessory]
+    public var accessories: [Accessory]
     internal var characteristicEventListeners: [Box<Characteristic>: WeakObjectSet<Server.Connection>]
     public var onIdentify: [(Accessory?) -> Void] = []
+    internal var configuration : Configuration
+    internal var onConfigurationChange: [(Device) -> Void] = []
+
+    // The device maitains a configuration number during its life time, which persists across restarts of the app
+
+    //    Current configuration number.
+    //    Must update when an accessory, service, or characteristic is added or removed on the accessory server.
+    //    Accessories must increment the config number after a firmware update.
+    //    This must have a range of 1-4294967295 and wrap to 1 when it overflows.
+    //    This value must persist across reboots, power cycles, etc.
+    //
+    //    Accessory instance IDs, "aid", are assigned from the same number pool that is global across
+    //    entire HAP Accessory Server. For example, if the first Accessory object has an instance ID of "1"
+    //    then no other Accessory object can have an instance ID of "1" within the Accessory Server.
+    //
+    //    Instance IDs are numbers with a range of [1, 18446744073709551615]. These numbers are used to
+    //    uniquely identify HAP accessory objects within an HAP accessory server, or uniquely identify
+    //    services, and characteristics within an HAP accessory object. The instance ID for each object
+    //    must be unique for the lifetime of the server/ client pairing.
+    
+    struct Configuration : Codable {
+        var number: UInt32 = 0
+        var lastAID: InstanceID = 1
```

Comment:
What should be the scope of this variable? We probably don't want this to be changed, outside of the generator and when initializing?

Reply:
This structure and all properties and func can be internal, and lastAID can be private